File: JavaCivitas/src/civitas/Casilla.java
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Daniel
 */
public class Casilla {
    
    public String nombre;

    Casilla(String in_nombre){

        nombre = in_nombre;
    }

    public void informe(int actual, ArrayList<Jugador> todos){

        Diario.getInstance().ocurreEvento("\n[DIARIO] El jugador "+String.valueOf(actual)+ " ha caido en "+this.toString());

    }

    void recibeJugador(int iactual, ArrayList<Jugador> todos){

        informe(iactual, todos);

    }

    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){

        // VERIFICAR SI EL INDICE DEL JUGADOR ES CORRECTO
        // comprueba si el primer
        // parámetro es un índice válido para acceder a los elementos del segundo parámetro.

        return actual >= 0 && actual < todos.size();
    }

    @Override
    public String toString(){

        return "\n[Casilla] -> "+ nombre + "\n[Tipo] -> Descanso\n";
    }
    
    public String getNombre(){
        
        return nombre;
        
    }
}

File: RubyCivitas/lib/casilla.rb
require_relative "diario"
require_relative "jugador"

module Civitas
  class Casilla

    def initialize(nombre)
      
      @nombre = nombre
      
    end
    
    def informe(actual, todos)

      Diario.instance.ocurre_evento("[DIARIO] El jugador #{todos[actual].nombre} ha caido en la casilla #{self.nombre.to_s}")

    end
    
    def jugador_correcto(actual, todos)

      correcto = false
      if actual < todos.size && actual >= 0
        correcto = true
      end

      return correcto
    end
    
    def recibe_jugador(actual, todos)
      informe(actual, todos)
    end
    
    #Override
    def to_string
      "[Casilla] -> #{@nombre}\n[Tipo] -> Descanso\n"
    end

    attr_reader :nombre
  end
end
